lib/compat/wordpress-6.2/default-filters.php

```diff
@@ -17,16 +17,24 @@
  * @package gutenberg
  */
 
-add_action( 'switch_theme', 'wp_theme_has_theme_json_clean_cache' );
-add_action( 'start_previewing_theme', 'wp_theme_has_theme_json_clean_cache' );
-add_action( 'upgrader_process_complete', '_wp_theme_has_theme_json_clean_cache_upon_upgrading_active_theme', 10, 2 );
-add_action( 'save_post_wp_global_styles', array( 'WP_Theme_JSON_Resolver_Gutenberg', 'clean_cached_data' ) );
+add_action( 'activated_plugin', 'gutenberg_get_global_settings_clean_cache' );
+add_action( 'activated_plugin', 'gutenberg_get_global_stylesheet_clean_cache' );
 add_action( 'activated_plugin', array( 'WP_Theme_JSON_Resolver_Gutenberg', 'clean_cached_data' ) );
+add_action( 'deactivated_plugin', 'gutenberg_get_global_settings_clean_cache' );
+add_action( 'deactivated_plugin', 'gutenberg_get_global_stylesheet_clean_cache' );
 add_action( 'deactivated_plugin', array( 'WP_Theme_JSON_Resolver_Gutenberg', 'clean_cached_data' ) );
-add_action( 'upgrader_process_complete', array( 'WP_Theme_JSON_Resolver_Gutenberg', '_clean_cached_data_upon_upgrading', 10, 2 ) );
+add_action( 'save_post_wp_global_styles', 'gutenberg_get_global_settings_clean_cache' );
 add_action( 'save_post_wp_global_styles', 'gutenberg_get_global_stylesheet_clean_cache' );
-add_action( 'switch_theme', 'gutenberg_get_global_stylesheet_clean_cache' );
+add_action( 'save_post_wp_global_styles', array( 'WP_Theme_JSON_Resolver_Gutenberg', 'clean_cached_data' ) );
+add_action( 'start_previewing_theme', 'gutenberg_get_global_settings_clean_cache' );
 add_action( 'start_previewing_theme', 'gutenberg_get_global_stylesheet_clean_cache' );
-add_action( 'activated_plugin', 'gutenberg_get_global_stylesheet_clean_cache' );
-add_action( 'deactivated_plugin', 'gutenberg_get_global_stylesheet_clean_cache' );
+add_action( 'start_previewing_theme', 'wp_theme_has_theme_json_clean_cache' );
```

Comment:
Are we sure we have all the cache invalidation here. What about mu-plugins or theme supports behind a option.

Reply:
This considers the same events as the stylesheet https://github.com/WordPress/gutenberg/pull/45679 What other events do you think we should consider for invalidating the cache?

Reply:
I didn't not review https://github.com/WordPress/gutenberg/pull/45679 😄 

- A add / remove theme supports in mu-plugin. 
- Adding data via a filter, in a plugin / theme / mu-plugin.

I am also worried about adding overhead to activation and deactivation of themes and plugins. These can already have fatel php in them.

Reply:
What are some mu-plugin events I should look at? The events for themes and plugins should be covered.

Reply:
There are no events for mu-plugins. There are not activated. 

> The events for themes and plugins should be covered.

What if my plugin has code like this 

```
if ( get_optoin( 'remove_theme_supports' ) === 1 ){
    remove_theme_supports( 'duotones');
}
```

How would you handle this case?

Reply:
There are a few options:

1. Consumers can always call the function to clean the cache, should they need. This can be done now.
2. We may want to introduce a specific filter to short-circuit the cache, as I proposed https://github.com/WordPress/gutenberg/pull/45679#discussion_r1026268881 That was punted to a follow-up, and it's now being discussed at https://github.com/WordPress/gutenberg/issues/45912

I haven't been able to dig deeper into how other cases (`WP_Theme` cache) handle this case.

Reply:
> Consumers can always call the function to clean the cache, should they need. This can be done now.

If they have to manually call these caches that is a breaking change and something we need to avoid. Extra context - https://github.com/WordPress/wordpress-develop/pull/3624#issuecomment-1323564738


The `wp_clean_themes_cache` function night be useful here. How WP_theme handles cache invalidation, is why I put together this. https://github.com/WordPress/wordpress-develop/pull/3604/

Reply:
3. We could consider cleaning the cache upon any option is added/updated/deleted. That would cover any dynamic behavior that depends on them. The performance will be still great given than a site is read many more times than any option is changed.

I think I'd only try this approach after we gather feedback on how this PR performs in the field.

Reply:
Options update, network option update and user meta could also be gates for if functionality is enabled. I don’t think it is realistic to cache clear on all these events. This cache clear has overhead and it will likely slow down all of WordPress if you do this.

Reply:
Have you any knowledge of how other parts of WordPress do this? I'd like to draw inspiration from other areas. So far, I've seen that `WP_Theme` implements one of the options I've proposed (the filter).

Reply:
> Have you any knowledge of how other parts of WordPress do this?

I'd like to help with this, but I'm not very familiar with WP caching.
Pingin @azaozz as he might know this better.

Reply:
I am the lead of the object caching team in the performance focus.

Reply:
> 3. We could consider cleaning the cache upon any option is added/updated/deleted. That would cover any dynamic behavior that depends on them. The performance will still be great given than reading a site is many times more frequent than changing an option.
> 
> I think I'd only try this approach after we gather feedback on how this PR performs in the field.

This third option seems a very good tradeoff. The are many plugins where the user has the option to enable or disable some blocks they offer, I presume this is stored in site options, so an invalidation on a site option change seems like a good thing to have.

Reply:
> I am the lead of the object caching team in the performance focus.

Yes, but the more eyes looking at the source code the better, isn't it?

Reply:
> Yes, but the more eyes looking at the source code the better, isn't it?

Yeah of course. Didn't say otherwise 😄